### PR TITLE
OHRI-443 reusable encounter list actions

### PR DESCRIPTION
**Pull request by pirupius**
This is a draft PR to clean up the encounter list component actions with multiple forms

### CODE REVIEW

File: src/components/encounter-list/encounter-list.component.tsx [hunk 1]
```diff
@@ -161,79 +161,35 @@ const EncounterList: React.FC<EncounterListProps> = ({
...
+                itemText={actionItem.label}
+                onClick={e => {
+                  e.preventDefault();
+                  if (actionItem.mode == 'edit') {
+                    launchFormInEditMode(
+                      applyFormIntent(actionItem.intent, getForm(actionItem.form.package, actionItem.form.name)),
+                      actionItem.encounterUuid,
+                      forceComponentUpdate,
+                    );
+                  } else {
```
Comment (samuelmale) [inline src/components/encounter-list/encounter-list.component.tsx:180]: Don't we have a case were we may want to launch a form in `enter` mode?

File: src/forms/components/inputs/_input.scss [hunk 2]
```diff
@@ -146,8 +146,8 @@
   color: $color-gray-70;
 }
 
-.errorLegend legend, .errorLabel label {
-  color: #da1e28
+.errorLabel > div label {
```
Comment (samuelmale) [inline src/forms/components/inputs/_input.scss:149]: Is this change related to the EncounterList enhancements?

**Review by samuelmale — CHANGES_REQUESTED** [review]
Looks good overall, left minor comments

File: src/forms/submission-handlers/base-handlers.ts [hunk 3]
```diff
@@ -68,7 +68,11 @@ export const ObsSubmissionHandler: SubmissionHandler = {
         );
       }
       if (typeof obs.value == 'string' || typeof obs.value == 'number') {
-        return field.questionOptions.rendering == 'date' ? moment(obs.value).toDate() : obs.value;
+        if (field.questionOptions.rendering == 'date') {
+          field.value.value = moment(field.value.value).format('YYYY-MM-DD HH:mm');
+          return moment(obs.value).toDate();
```
Comment (samuelmale) [inline src/forms/submission-handlers/base-handlers.ts:73]: whats happening here?

**Review by ebambo — CHANGES_REQUESTED** [review]

File: src/forms/components/inputs/_input.scss (hunk 2 above)
Comment (pirupius) [reply src/forms/components/inputs/_input.scss:149]: The covid feature branch is behind dev, a rebase will fix this

File: src/forms/submission-handlers/base-handlers.ts (hunk 3 above)
Comment (pirupius) [reply src/forms/submission-handlers/base-handlers.ts:73]: Will be fixed with the update of the rebase

File: src/components/encounter-list/encounter-list.component.tsx [hunk 4]
```diff
@@ -161,79 +161,35 @@ const EncounterList: React.FC<EncounterListProps> = ({
         row[column.key] = val;
       });
 
-      //TODO: This 'piece of hack' SHOULD BE REMOVED once Pius is done with the Actions implementation
-      if (form.package == 'covid' && form.name == 'covid_assessment') {
+      if (row['actions']) {
+        let actionItems = row['actions'];
+        delete row['actions'];
```
[the remark this answers is not in the document]
Comment (pirupius) [reply src/components/encounter-list/encounter-list.component.tsx:166]: Was having some reassignment issues with this key and resulted to this. Will take another spin to look for an alternative

File: src/forms/components/inputs/_input.scss [hunk 5]
```diff
@@ -6,7 +6,7 @@
 }
 
 .formField > div > div > label {
-  color: #525252 !important;
```
[the remark this answers is not in the document]
Comment (pirupius) [reply src/forms/components/inputs/_input.scss:9]: The covid feature branch is behind dev, will rebase the feature branch to fix this

File: src/forms/submission-handlers/base-handlers.ts [hunk 6]
```diff
@@ -68,7 +68,11 @@ export const ObsSubmissionHandler: SubmissionHandler = {
         );
       }
       if (typeof obs.value == 'string' || typeof obs.value == 'number') {
-        return field.questionOptions.rendering == 'date' ? moment(obs.value).toDate() : obs.value;
+        if (field.questionOptions.rendering == 'date') {
```
[the remark this answers is not in the document]
Comment (pirupius) [reply src/forms/submission-handlers/base-handlers.ts:71]: Will be resolved by the rebase as well

File: src/components/encounter-list/encounter-list.component.tsx (hunk 1 above)
Comment (pirupius) [reply src/components/encounter-list/encounter-list.component.tsx:180]: Am not sure about this for encounter actions. Will look into it to be sure

File: src/components/encounter-list/encounter-list.component.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (pirupius) [reply src/components/encounter-list/encounter-list.component.tsx:166]: Was having some reassignment issues here

File: src/components/encounter-list/encounter-list.component.tsx [hunk 7]
```diff
@@ -161,79 +161,35 @@ const EncounterList: React.FC<EncounterListProps> = ({
         row[column.key] = val;
       });
 
-      //TODO: This 'piece of hack' SHOULD BE REMOVED once Pius is done with the Actions implementation
-      if (form.package == 'covid' && form.name == 'covid_assessment') {
+      if (row['actions']) {
+        let actionItems = row['actions'];
```
[the remark this answers is not in the document]
Comment (pirupius) [reply src/components/encounter-list/encounter-list.component.tsx:165]: 😅

**Review by samuelmale — APPROVED** [review]
LGTM, thanks Pius
